ompi/mca/coll/acoll/coll_acoll_utils.h

```diff
@@ -73,6 +77,95 @@ static inline void coll_acoll_buf_free(coll_acoll_reserve_mem_t *reserve_mem_ptr
     }
 }
 
+/* Function to check if subcomms structure is allocated and initialized */
+static inline int check_and_create_subc(ompi_communicator_t *comm,
+                                        mca_coll_acoll_module_t *acoll_module,
+                                        coll_acoll_subcomms_t **subc_ptr)
+{
+    int cid = ompi_comm_get_local_cid(comm);
+    int num_subc = acoll_module->num_subc;
+    coll_acoll_subcomms_t *subc;
+
+    /* Return if max comms is not positive */
+    if (acoll_module->max_comms <= 0) {
+        *subc_ptr = NULL;
+        return MPI_SUCCESS;
+    }
+
+    /* Allocate memory for subcomms array */
+    if (acoll_module->subc == NULL) {
+        acoll_module->subc = malloc(sizeof(coll_acoll_subcomms_t**) * acoll_module->max_comms);
+        if (acoll_module->subc == NULL) {
+            return OMPI_ERR_OUT_OF_RESOURCE;
+        }
+    }
+
+    /* Check if subcomms structure is already created for the communicator */
+    for (int i = 0; i < num_subc; i++) {
+        if (acoll_module->subc[i]->cid == cid) {
+            *subc_ptr = acoll_module->subc[i];
+            return MPI_SUCCESS;
+        }
+    }
+
+    /* Subcomms structure is not present, create one if within limit*/
+    if (num_subc == acoll_module->max_comms) {
+        *subc_ptr = NULL;
```

Comment:
could we add here an opal_output() warning to the user we are using the fallback path and suggest that he can increase the max no. of communicators handled by the component using the mca parameter? This warning could be displayed with verbosity level turned on or similar.

Reply:
Added.